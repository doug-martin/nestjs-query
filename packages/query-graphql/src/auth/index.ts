export * from './authorizer';
export * from './default-crud.authorizer';
export * from './tokens';
