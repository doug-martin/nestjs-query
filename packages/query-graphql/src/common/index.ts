export { DTONamesOpts, getDTONames, DTONames, getDTOIdTypeOrDefault } from './dto.utils';
export * from './external.utils';
export * from './resolver.utils';
export * from './object.utils';
