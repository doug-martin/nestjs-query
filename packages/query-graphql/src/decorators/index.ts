export {
  FilterableField,
  FilterableFieldOptions,
  FilterableFieldDescriptor,
  getFilterableFields,
} from './filterable-field.decorator';
export { ResolverMethodOpts } from './resolver-method.decorator';
export {
  CursorConnection,
  FilterableCursorConnection,
  OffsetConnection,
  FilterableOffsetConnection,
  UnPagedRelation,
  FilterableUnPagedRelation,
  Relation,
  FilterableRelation,
  RelationDecoratorOpts,
  RelationTypeFunc,
  getRelations,
} from './relation.decorator';
export * from './resolver-mutation.decorator';
export * from './resolver-query.decorator';
export * from './resolver-field.decorator';
export { Reference, ReferenceDecoratorOpts, ReferenceTypeFunc } from './reference.decorator';
export { ResolverSubscription, SubscriptionResolverMethodOpts } from './resolver-subscription.decorator';
export { InjectPubSub } from './inject-pub-sub.decorator';
export * from './skip-if.decorator';
export * from './aggregate-query-param.decorator';
export * from './hook.decorator';
export * from './decorator.utils';
export * from './hook-args.decorator';
export * from './authorizer.decorator';
export * from './inject-authorizer.decorator';
export * from './inject-custom-authorizer.decorator';
export * from './key-set.decorator';
export * from './authorize-filter.decorator';
export * from './query-options.decorator';
export * from './id-field.decorator';
