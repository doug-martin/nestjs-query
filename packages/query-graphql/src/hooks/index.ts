export * from './hooks';
export { createDefaultHook } from './default.hook';
export * from './tokens';
export * from './types';
