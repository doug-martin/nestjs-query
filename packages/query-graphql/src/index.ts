export * from './types';
export {
  FilterableField,
  FilterableFieldOptions,
  ResolverMethodOpts,
  Relation,
  FilterableRelation,
  CursorConnection,
  FilterableCursorConnection,
  OffsetConnection,
  FilterableOffsetConnection,
  UnPagedRelation,
  FilterableUnPagedRelation,
  RelationTypeFunc,
  RelationDecoratorOpts,
  Reference,
  ReferenceTypeFunc,
  ReferenceDecoratorOpts,
  InjectPubSub,
  BeforeCreateOne,
  BeforeCreateMany,
  BeforeUpdateOne,
  BeforeUpdateMany,
  BeforeDeleteOne,
  BeforeDeleteMany,
  BeforeQueryMany,
  BeforeFindOne,
  InjectAuthorizer,
  InjectCustomAuthorizer,
  Authorize,
  AuthorizerFilter,
  RelationAuthorizerFilter,
  HookArgs,
  MutationHookArgs,
  KeySet,
  QueryOptions,
  IDField,
  IDFieldOptions,
} from './decorators';
export * from './resolvers';
export * from './federation';
export { DTONamesOpts } from './common';
export { NestjsQueryGraphQLModule } from './module';
export { AutoResolverOpts } from './providers';
export { pubSubToken, GraphQLPubSub } from './subscription';
export { Authorizer, CustomAuthorizer, AuthorizerOptions, AuthorizationContext, OperationGroup } from './auth';
export {
  Hook,
  HookTypes,
  BeforeCreateOneHook,
  BeforeCreateManyHook,
  BeforeUpdateOneHook,
  BeforeUpdateManyHook,
  BeforeDeleteOneHook,
  BeforeDeleteManyHook,
  BeforeQueryManyHook,
  BeforeFindOneHook,
} from './hooks';
export { AuthorizerInterceptor, AuthorizerContext, HookInterceptor, HookContext } from './interceptors';
