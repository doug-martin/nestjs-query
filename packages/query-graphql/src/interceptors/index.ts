export * from './hook.interceptor';
export * from './authorizer.interceptor';
