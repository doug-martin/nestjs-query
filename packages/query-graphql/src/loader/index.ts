export * from './find-relations.loader';
export * from './query-relations.loader';
export * from './count-relations.loader';
export * from './aggregate-relations.loader';
export * from './dataloader.factory';
