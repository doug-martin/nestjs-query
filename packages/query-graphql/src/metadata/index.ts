import { GraphQLQueryMetadataStorage } from './metadata-storage';

export { FilterableFieldDescriptor } from './metadata-storage';

/** @internal */
let storage: GraphQLQueryMetadataStorage;
/** @internal */
export function getMetadataStorage(): GraphQLQueryMetadataStorage {
  if (!storage) {
    storage = new GraphQLQueryMetadataStorage();
  }
  return storage;
}
