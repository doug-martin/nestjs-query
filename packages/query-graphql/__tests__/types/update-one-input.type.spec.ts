// eslint-disable-next-line max-classes-per-file
import { plainToClass } from 'class-transformer';
import { validateSync, MinLength } from 'class-validator';
import { InputType, Resolver, Args, Field, Query, Int, ID, ObjectType } from '@nestjs/graphql';
import { IDField, UpdateOneInputType } from '../../src';
import { generateSchema } from '../__fixtures__';

describe('UpdateOneInputType', (): void => {
  @ObjectType()
  class FakeDTO {
    @Field(() => ID)
    id!: string;
  }

  @InputType()
  class FakeUpdateOneType {
    @Field()
    @MinLength(5)
    name!: string;
  }

  @InputType()
  class UpdateOne extends UpdateOneInputType(FakeDTO, FakeUpdateOneType) {}

  it('should create an input type with the id and update type as fields', async () => {
    @Resolver()
    class UpdateOneInputTypeSpec {
      @Query(() => Int)
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      updateTest(@Args('input') input: UpdateOne): number {
        return 1;
      }
    }
    const schema = await generateSchema([UpdateOneInputTypeSpec]);
    expect(schema).toMatchSnapshot();
  });

  it('should create an input type with a custom id and update type as fields', async () => {
    @ObjectType()
    class FakeIDDTO {
      @IDField(() => String)
      id!: string;
    }

    @InputType()
    class UpdateOneCustomId extends UpdateOneInputType(FakeIDDTO, FakeUpdateOneType) {}

    @Resolver()
    class UpdateOneCustomIdInputTypeSpec {
      @Query(() => Int)
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      updateTest(@Args('input') input: UpdateOneCustomId): number {
        return 1;
      }
    }
    const schema = await generateSchema([UpdateOneCustomIdInputTypeSpec]);
    expect(schema).toMatchSnapshot();
  });

  describe('validation', () => {
    it('should validate id is defined is not empty', () => {
      const Type = UpdateOneInputType(FakeDTO, FakeUpdateOneType);
      const input = { update: { name: 'hello world' } };
      const it = plainToClass(Type, input);
      const errors = validateSync(it);
      expect(errors).toEqual([
        {
          children: [],
          constraints: {
            isNotEmpty: 'id should not be empty',
          },
          property: 'id',
          target: input,
        },
      ]);
    });

    it('should validate id is not empty is defined is not empty', () => {
      const Type = UpdateOneInputType(FakeDTO, FakeUpdateOneType);
      const input = { id: '', update: { name: 'hello world' } };
      const it = plainToClass(Type, input);
      const errors = validateSync(it);
      expect(errors).toEqual([
        {
          children: [],
          constraints: {
            isNotEmpty: 'id should not be empty',
          },
          property: 'id',
          target: input,
          value: input.id,
        },
      ]);
    });

    it('should validate the update input', () => {
      const Type = UpdateOneInputType(FakeDTO, FakeUpdateOneType);
      const input = { id: 'id-1', update: {} };
      const it = plainToClass(Type, input);
      const errors = validateSync(it);
      expect(errors).toEqual([
        {
          children: [
            {
              children: [],
              constraints: {
                minLength: 'name must be longer than or equal to 5 characters',
              },
              property: 'name',
              target: {},
            },
          ],
          property: 'update',
          target: it,
          value: it.update,
        },
      ]);
    });
  });
});
