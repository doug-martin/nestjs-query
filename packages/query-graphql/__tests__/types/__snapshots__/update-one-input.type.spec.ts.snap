// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateOneInputType should create an input type with a custom id and update type as fields 1`] = `
type Query {
  updateTest(input: UpdateOneCustomId!): Int!
}

input UpdateOneCustomId {
  """The id of the record to update"""
  id: String!

  """The update to apply."""
  update: FakeUpdateOneType!
}

input FakeUpdateOneType {
  name: String!
}

`;

exports[`UpdateOneInputType should create an input type with the id and update type as fields 1`] = `
type Query {
  updateTest(input: UpdateOne!): Int!
}

input UpdateOne {
  """The id of the record to update"""
  id: ID!

  """The update to apply."""
  update: FakeUpdateOneType!
}

input FakeUpdateOneType {
  name: String!
}

`;
